### PR TITLE
[ibex/dv] add Questa support

### PR DESCRIPTION
**Pull request by bp-ouhaha**
my trial to add Mentor QuestaSim support for ibex_core regression

Signed-off-by: Bert Pieters <bert.pieters@gmail.com>

### CODE REVIEW

File: dv/uvm/core_ibex/yaml/rtl_simulation.yaml [hunk 1]
```diff
@@ -52,6 +52,29 @@
     wave_opts: >
       -ucli -do <cwd>/vcs.tcl
 
+- tool: questa
+  compile:
+    cmd:
+      - "vmap mtiUvm $QUESTA_HOME/questasim/uvm-1.2"
+      - "vlog -64
+        -access=rwc
```
Comment (imphil) [inline dv/uvm/core_ibex/yaml/rtl_simulation.yaml:60]: Can you indent this line a bit more to follow the style of the xlm target for example?

Comment (bp-ouhaha) [reply dv/uvm/core_ibex/yaml/rtl_simulation.yaml:60]: ok will do

File: dv/uvm/core_ibex/sim_makefrag_gen.py [hunk 2]
```diff
@@ -79,7 +94,7 @@ def main():
         'Generates a makefile fragment for use with the Ibex DV makefile that '
         'sets up sim specific variables'))
 
-    sim_fns = {'vcs': gen_vcs_makefrag, 'riviera': gen_riviera_makefrag, 'xlm': gen_xlm_makefrag}
+    sim_fns = {'vcs': gen_vcs_makefrag, 'riviera': gen_riviera_makefrag, 'xlm': gen_xlm_makefrag, 'questa': gen_questa_makefrag}
```
Comment (GregAC) [inline dv/uvm/core_ibex/sim_makefrag_gen.py:97]: One minor thing: We aim to keep line sizes short in python (79 characters or less as specified by PEP8 https://www.python.org/dev/peps/pep-0008/#maximum-line-length) so could you break this line up?

You weren't to know as we don't actually document our preferred Python style in the Ibex repository as far as I can tell! The OpenTitan documentation has the full details: https://docs.opentitan.org/doc/rm/python_coding_style/

Comment (bp-ouhaha) [reply dv/uvm/core_ibex/sim_makefrag_gen.py:97]: will break it up, one per line

File: dv/uvm/core_ibex/yaml/rtl_simulation.yaml (hunk 1 above)
Comment (bp-ouhaha) [reply dv/uvm/core_ibex/yaml/rtl_simulation.yaml]: it's updated accordingly

File: dv/uvm/core_ibex/sim_makefrag_gen.py (hunk 2 above)
Comment (bp-ouhaha) [reply dv/uvm/core_ibex/sim_makefrag_gen.py]: it's updated accordingly

File: dv/uvm/core_ibex/sim_makefrag_gen.py [hunk 3]
```diff
@@ -79,7 +94,10 @@ def main():
         'Generates a makefile fragment for use with the Ibex DV makefile that '
         'sets up sim specific variables'))
 
-    sim_fns = {'vcs': gen_vcs_makefrag, 'riviera': gen_riviera_makefrag, 'xlm': gen_xlm_makefrag}
+    sim_fns = {'vcs'    : gen_vcs_makefrag,
+               'riviera': gen_riviera_makefrag,
+               'xlm'    : gen_xlm_makefrag,
+               'questa' : gen_questa_makefrag}
```
Comment (rswarbrick) [inline dv/uvm/core_ibex/sim_makefrag_gen.py:100]: Sorry, I realise this is really nitty, but I think PEP8 standard style doesn't allow tabulating dicts like this. Could you drop the extra spaces?

Comment (rswarbrick) [reply dv/uvm/core_ibex/sim_makefrag_gen.py:100]: If you run flake8 on the file, you should hopefully get no errors. We've got that check in OpenTitan CI, but apparently not in Ibex.

Comment (bp-ouhaha) [reply dv/uvm/core_ibex/sim_makefrag_gen.py]: I've fixed it and run it through flake8 (didn't know that tool)

**Comment by imphil** [comment]
Thanks for the updates @bp-ouhaha. The one remaining question is about `qrun`: Do you know the difference between that and Questa? Is it a different way of calling it?

**Comment by bp-ouhaha** [comment]
> Thanks for the updates @bp-ouhaha. The one remaining question is about `qrun`: Do you know the difference between that and Questa? Is it a different way of calling it?

I don't know, seen the syntax, snapshot word, and in fyle qrun_option.f, it looks cadence-alike.

**Comment by imphil** [comment]
OK, let's ignore the `qrun` issue for the moment :-) Looks good, merging.

**Pull request merged**